addons/website_slides/tests/test_partner_merge.py

```diff
@@ -0,0 +1,146 @@
+                    'sequence': 1,
+                    'slide_category': 'document',
+                    'is_published': True,
+                    'is_preview': True,
+                })],
+        })
+        cls.channel.karma_gen_channel_finish = 15
+        cls.channel2_slide1 = cls.channel2.slide_ids[0]
+        cls.user_students = [
+            mail_new_test_user(
+                cls.env,
+                email=f'student{student_n}@example.com',
+                groups='base.group_portal',
+                login=f'user_student{student_n}',
+                name=f'Student {student_n}',
+                notification_type='email',
+            ) for student_n in range(3)]
+        cls.user_student0, cls.user_student1, cls.user_student2 = cls.user_students
+        # Initial karma
+        cls.user_student0.karma = 100
+        cls.user_student1.karma = 200
+        cls.user_student2.karma = 300
+        # Initial courses/slides membership and completion
+        cls.join_course(cls.user_student0, cls.channel)
+        cls.join_course(cls.user_student0, cls.channel2)
+        cls.join_course(cls.user_student1, cls.channel)
+        cls.join_course(cls.user_student2, cls.channel)
+        cls.complete_slide(cls.user_student0, cls.slide)
+        cls.complete_slide(cls.user_student0, cls.channel2_slide1)  # karma 10 for completing the channel (1 slide)
+        cls.complete_slide(cls.user_student1, cls.slide_3)  # karma 42
+        cls.complete_slide(cls.user_student2, cls.slide_2)
+
+    def merge_partners(self, partner_ids):
+        self.MergeWizard._merge(partner_ids, self.env['res.partner'].browse(partner_ids[-1]))
+
+    @classmethod
+    def complete_slide(cls, user_stud, slide):
+        slide_with_stud = slide.with_user(user_stud)
+        if slide.question_ids:
+            slide_with_stud.action_set_viewed(quiz_attempts_inc=True)
+        slide_with_stud._action_mark_completed()
+
+    @classmethod
+    def join_course(cls, user_stud, channel):
+        channel._action_add_members(user_stud.partner_id)
+
+    def test_initial_values(self):
+        self.assertEqual(self.user_student0.karma, 100 + 10)
+        self.assertEqual(self.user_student1.karma, 200 + 42)
+        self.assertEqual(self.user_student2.karma, 300)
+        self.assertEqual(len(self.channel.partner_ids), 4)
+        self.assertEqual(len(self.channel2.partner_ids), 2)
+
+    def test_merge_0_partner(self):
+        self.MergeWizard._merge(self.env['res.users'].ids)
+        self.test_initial_values()  # No change must occur
+
+    def test_merge_1_partner(self):
+        self.MergeWizard._merge(self.user_student0.ids)
+        self.test_initial_values()  # No change must occur
+
+    def test_merge_2_partners(self):
+        # for partner_ids_to_merge in permutations([u.partner_id.id for u in self.user_students[:2]]):
+        for partner_ids_to_merge in permutations([u.partner_id.id for u in self.user_students[:2]]):
```

Comment:
I think all run after the first one will be useless
![screenshot_spect](https://user-images.githubusercontent.com/54316226/234599368-b2c91ca0-6b88-4e36-8ded-39abd1d6ba50.png)